From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 27 Sep 2010 20:34:52 -0400
Subject: [net] qlcnic: fix gro support
Message-id: <20100927203452.3859.88061.sendpatchset@localhost.localdomain>
Patchwork-id: 28430
O-Subject: [RHEL 5.6 PATCH 1/6] qlcnic: fix gro support
Bugzilla: 625084
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla
--------

Bug 625084 (https://bugzilla.redhat.com/show_bug.cgi?id=625084)

Upstream Status
---------------

RHEL 5 specific.

Description
-----------

>From efcdbe43fd69898ac02d04820862815225cdb4f3 Mon Sep 17 00:00:00 2001
From: Amit Kumar Salecha <amit.salecha@qlogic.com>
Date: Wed, 21 Jul 2010 02:26:57 -0700
Subject: [PATCH] qlcnic: fix gro support

GRO capability was not announced in netdev features.

diff --git a/drivers/net/qlcnic/qlcnic_main.c b/drivers/net/qlcnic/qlcnic_main.c
index 238c1fe..174551e 100644
--- a/drivers/net/qlcnic/qlcnic_main.c
+++ b/drivers/net/qlcnic/qlcnic_main.c
@@ -1283,14 +1283,14 @@ qlcnic_setup_netdev(struct qlcnic_adapter *adapter,
 	netdev->poll_controller = qlcnic_poll_controller;
 #endif
 
-	netdev->features |= (NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_HW_CSUM);
+	netdev->features |= (NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_HW_CSUM |
+				NETIF_F_GRO);
 
 	if (adapter->capabilities & QLCNIC_FW_CAPABILITY_TSO)
 		netdev->features |= (NETIF_F_TSO | NETIF_F_TSO6);
 
-	if (pci_using_dac) {
+	if (pci_using_dac)
 		netdev->features |= NETIF_F_HIGHDMA;
-	}
 
 	if (adapter->capabilities & QLCNIC_FW_CAPABILITY_FVLANTX)
 		netdev->features |= (NETIF_F_HW_VLAN_TX);
